shard/src/main/java/com/redhat/developer/shard/controllers/ProcessorController.java

```diff
@@ -0,0 +1,76 @@
+import com.redhat.developer.shard.ManagerSyncService;
+
+@ApplicationScoped
+public class ProcessorController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProcessorController.class);
+
+    private List<ProcessorDTO> processors = new ArrayList<>();
+
+    @Inject
+    ManagerSyncService managerSyncService;
+
+    @Inject
+    ExecutorsService executorsService;
+
+    public void reconcileProcessorsFor(BridgeDTO bridgeDTO) {
+        managerSyncService.fetchProcessorsForBridge(bridgeDTO)
+                .subscribe()
+                .with(item -> reconcileProcessor(item),
+                        failure -> LOG.error("Failed to retrieve list of Processors from Manager for Bridge '{}'", bridgeDTO.getId()),
+                        () -> LOG.info("Initiated Reconcile for all Processors for bridge '{}' for customer '{}'.", bridgeDTO.getId(), bridgeDTO.getCustomerId()));
+    }
+
+    private void failedToSendUpdateToManager(ProcessorDTO dto, Throwable error) {
+        LOG.error("Failed to send status update to Manager for Processor '{}' on Bridge '{}'", dto.getId(), dto.getBridge().getId(), error);
+    }
+
+    private void createExecutor(ProcessorDTO processorDTO) {
+        try {
+            executorsService.createExecutor(processorDTO);
+            processorDTO.setStatus(BridgeStatus.AVAILABLE);
+        } catch (Throwable t) {
+            LOG.error("Failed to reconcile Executor for Processor '{}' for customer '{}' on Bridge '{}'", processorDTO.getId(), processorDTO.getBridge().getCustomerId(),
+                    processorDTO.getBridge().getId(), t);
+            processorDTO.setStatus(BridgeStatus.FAILED);
+        }
+        managerSyncService.notifyProcessorStatusChange(processorDTO).subscribe()
+                .with(success -> LOG.info("Completed reconcile of Executor for Processor '{}' on Bridge '{}'. Final status: '{}'", processorDTO.getId(), processorDTO.getBridge().getId(),
+                        processorDTO.getStatus()),
+                        failure -> failedToSendUpdateToManager(processorDTO, failure));
+    }
+
+    private void reconcileProcessor(ProcessorDTO processorDTO) {
+
+        if (processorDTO.getStatus() == BridgeStatus.REQUESTED) {
+
+            processorDTO.setStatus(BridgeStatus.PROVISIONING);
+            this.processors.add(processorDTO);
+            managerSyncService.notifyProcessorStatusChange(processorDTO)
+                    .subscribe()
+                    .with(item -> createExecutor(processorDTO),
+                            failure -> failedToSendUpdateToManager(processorDTO, failure));
+        } else if (processorDTO.getStatus() == BridgeStatus.PROVISIONING) {
+            /*
+             * If the Manager is still reporting the Processor as provisioning and a new reconcile loop starts, try to provision again. This operation is
+             * idempotent
+             */
+            createExecutor(processorDTO);
+        }
```

Comment:
I think it would be better to move this logic to something separated -> even with a different endpoint from the shard. 

We have to investigate a bit what is the failure strategy, for example: 
1) the istance is not deployed -> deploy it
2) the instance is already deployed and running -> update the status
3) the instance is in a failure state -> report the state

Reply:
@r00ta The only goal here currently was to ensure that we attempt to provision the Executor if we "rebooted" the Shard  after it had sent the `PROVISIONING` to the Manager but before it had got around to deploying the Executor.

In hindsight, it doesn't really work because we'd lose all `AVAILABLE` Executors and not attempt to re-create them, so I'll remove this.